src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php

```diff
@@ -76,5 +80,21 @@ protected function initialize()
                 $this->addLoader($alias, $this->container->get($id));
             }
         }
+
+        if ($this->resourceDirs) {
+            $finder = Finder::create()
+                ->files()
+                ->filter(function (\SplFileInfo $file) {
+                    return 2 === substr_count($file->getBasename(), '.') && preg_match('/\.\w+$/', $file->getBasename());
+                })
+                ->in($this->resourceDirs)
+            ;
+
+            foreach ($finder as $file) {
+                // filename is domain.locale.format
+                list($domain, $locale, $format) = explode('.', $file->getBasename(), 3);
+                $this->addResource($format, (string) $file, $locale, $domain);
```

Comment:
Here, I think we can do even better as we could only load the relevant files for the locale we are working on. To make it work, we would need to pass the locale to this method, which is not possible for BC reasons, but as the `initialize` method is only called once, what about moving this code into a new private method called after `initialize` in `loadCatalogue`?